### PR TITLE
Support subscription level deployment commands.

### PR DESCRIPTION
**Pull request by Tiano2017**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by lmazuel** [comment]
@Tiano2017 Your package is released on PyPI. Please:
- Remove the "privates" folder.
- Update ALL "setup.py" that contains a reference to "azure-mgmt-resource==1.2.1" to "azure-mgmt-resource==2.0.0rc2"

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py [hunk 1]
```diff
@@ -144,6 +144,21 @@ def load_arguments(self, _):
     with self.argument_context('group deployment operation show') as c:
         c.argument('operation_ids', nargs='+', help='A list of operation ids to show')
 
+    with self.argument_context('deployment') as c:
+        c.argument('deployment_name', options_list=('--name', '-n'), required=True, help='The deployment name.')
+        c.argument('deployment_location', options_list=('--location', '-l'), required=True, help='The deployment location.')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py]: options_list should use list syntax, not tuple.

Comment (TianoMS) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py]: Not quite understand what the issue is.. can you give more details? I'm following what we do for "group deployment".

---
In reply to: [195173832](https://github.com/Azure/azure-cli/pull/6572#discussion_r195173832) [](ancestors = 195173832)

Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py]: It should look like `options_list=['--location', '-l']`

**Comment by TianoMS** [comment]
>             time.sleep(5)

to be removed...

---
Refers to: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/latest/test_resource.py:221 in 44d66c0. [](commit_id = 44d66c05150c79259af6787018f4b37366cd35cb, deletion_comment = False)

**Comment by TianoMS** [comment]
>         self.cmd('deployment show -n {dn}', checks=[

add a check for export. 

---
Refers to: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/tests/latest/test_resource.py:326 in 44d66c0. [](commit_id = 44d66c05150c79259af6787018f4b37366cd35cb, deletion_comment = False)

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py [hunk 2]
```diff
@@ -144,6 +144,21 @@ def load_arguments(self, _):
...
+        c.argument('template_file', completer=FilesCompleter(), type=file_type, help="a template file path in the file system")
+        c.argument('template_uri', help='a uri to a remote template file')
+        c.argument('mode', arg_type=get_enum_type(DeploymentMode, default='incremental'), help='Only Incremental is supported for deployment at subscription scope')
```
Comment (TianoMS) [inline src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py]: >mode [](start = 20, length = 4)

we can get rid of this parameter.

File: src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py (hunk 1 above)
Comment (tjprescott) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py]: Also, here's the metadata from the global location alias:
```Python
    location_type = CLIArgumentType(
        options_list=('--location', '-l'),
        completer=get_location_completion_list,
        type=get_location_name_type(cli_ctx),
        help="Location. You can configure the default location using `az configure --defaults location=<location>`",
        metavar='LOCATION',
        configured_default='location')
```

You should register your param like this:
```Python
c.argument('deployment_location', arg_type=get_location_type(self.cli_ctx), required=True)
```

If your command signature does not show a default value for location, you don't even need the `required=True` part. The CLI will automatically know the location is required.

Comment (Tiano2017) [reply src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/_params.py]: thanks. I've updated this one.

**Comment by troydai** [comment]
Please hold on the changes. I will bump the core version in another PR.

**Comment by Tiano2017** [comment]
got it. let me know when it's done. Do I need to revert any change?

**Comment by troydai** [comment]
You don't need to revert any changes. Just rebase on top of the dev.

**Comment by troydai** [comment]
Please rebase on to latest dev

**Comment by Tiano2017** [comment]
@troydai rebase done.

**Comment by troydai** [comment]
Thank you, @Tiano2017 .

**Comment by Tiano2017** [comment]
Thanks @troydai  and @yugangw-msft for the help on this PR.

**Pull request merged**